src/browser.js

```diff
@@ -19,25 +21,27 @@ export default function reactExpressMiddlewareGenerator (options = {}) {
 
 	return function reactExpressMiddleware (req, res, next) {
 		res.renderReactComponent = function renderReactComponent (Component, store, done) {
-			// store defaults to res.locals
 			if (typeof store === 'function') {
 				done = store;
 				store = undefined;
 			}
-			if (typeof store !== 'object') {
-				store = res.locals;
 
-				// Override prototype on res.locals because react
-				// uses res.locals.hasOwnProperty and Express specefically
-				// uses Object.create(null) which means it doesn't have hasOwnProperty
-				setPrototypeOf(store, {});
-			}
+			if (!Component.$$typeof) {
```

Comment:
We should use `Object.hasOwnProperty` here, or at the minimum use `typeof Component.$$typeof !== 'undefined'`. Any reliance on explicitly internal stuff like this is frowned upon and needs to be handled with care.